dask_kubernetes/operator/kubecluster/kubecluster.py

```diff
@@ -230,6 +230,23 @@ def __init__(
         if isinstance(self.worker_command, str):
             self.worker_command = self.worker_command.split(" ")
 
+        if self.n_workers is not None and not isinstance(self.n_workers, int):
+            raise TypeError(f"n_workers must be an integer, got {type(self.n_workers)}")
+
+        try:
+            # Validate `resources` param is a dictionary whose
+            # keys must either be 'limits' or 'requests'
+            assert isinstance(self.resources, dict)
+
+            for field in self.resources:
+                if field in ("limits", "requests"):
+                    assert isinstance(self.resources[field], dict)
+                else:
+                    raise ValueError(f"Unknown field '{field}' in resources")
+        except TypeError:
+            print(f"invalid '{type(resources)}' for resources type")
+            raise
```

Comment:
Let's not print here. Printing is usually not a good idea. If you want to explicitly set the error message maybe raise a new error from the caught one?

```suggestion
        except TypeError as e:
            raise TypeError(f"invalid '{type(resources)}' for resources type") from e
```